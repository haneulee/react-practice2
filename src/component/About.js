import React, { Component } from "react";

class About extends Component {
  constructor(props) {
    super(props);
    this.state = {
      seconds: Date.now() / 1000
    };
  }
  componentDidMount() {
    this.timerID = setInterval(() => this.tick(), 1000);
  }
  componentWillUnmount() {
    clearInterval(this.timerID);
  }
  tick() {
    this.setState({ seconds: Date.now() / 1000 });
  }
  goHome() {
    if (window.confirm("do you really want to go Home?") === true) {
      this.props.history.push("/");
    }
  }
  render() {
    return (
      <div>
        <h2>About</h2>
        <p>{this.state.seconds} seconds have ellapsed since the UNIX epoch.</p>
        <button className="btn btn-primary" onClick={this.goHome.bind(this)}>
          home
        </button>
      </div>
    );
  }
}

export default About;
