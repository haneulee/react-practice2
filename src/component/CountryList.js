import React, { Component } from "react";
import CountryItem from "./CountryItem";
import PropTypes from "prop-types"

class CountryList extends Component {
  render() {
    let countries = this.props.itemlist.map((item, index) => {
      return <CountryItem key={item.no} {...item} />;
    });
    return <ul className="list-group">{countries}</ul>;
  }
}

CountryList.propTypes = {
    itemlist: PropTypes.arrayOf(PropTypes.object)
}

export default CountryList;
