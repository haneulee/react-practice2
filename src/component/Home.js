import React, { Component } from "react";

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isChecked: false
    };
    this.onChange = this.onChange.bind(this);
  }
  onChange() {
    this.setState({ isChecked: !this.state.isChecked });
  }
  render() {
    return (
      <div>
        {/* <h2>Home</h2> */}
        <label>
          <input
            type="checkbox"
            checked={this.state.isChecked}
            onChange={this.onChange}
          />
          {this.state.isChecked ? this.props.labelOn : this.props.labelOff}
        </label>
      </div>
    );
  }
}

export default Home;
