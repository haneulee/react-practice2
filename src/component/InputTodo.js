import React, { Component } from "react";
import PropTypes from "prop-types";
import Logger from "./Logger";

class InputTodo extends Component {
  constructor() {
    super();
    this.state = {
      todo: ""
    };
  }
  addHandler = e => {
    this.props.add(this.state.todo);
    this.setState({ todo: "" });
  };
  changeTodo = e => {
    this.setState({ todo: e.target.value });
  };
  render() {
    return (
      <div className="row">
        <div className="col">
          <div className="input-group">
            <input
              id="msg"
              type="text"
              className="form-control"
              name="msg"
              placeholder="to do..."
              value={this.state.todo}
              onChange={this.changeTodo}
            />
            <span
              className="btn btn-primary input-group-addon"
              onClick={this.addHandler}
            >
              추가
            </span>
          </div>
        </div>
      </div>
    );
  }
}

InputTodo.propTypes = {
  add: PropTypes.func.isRequired
};

export default Logger(InputTodo);
