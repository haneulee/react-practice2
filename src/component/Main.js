import { Component } from "react";
import { renderRoutes } from "react-router-config";
import { routes } from "../routes";

class Main extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    return renderRoutes(routes, this.props);
  }
}

export default Main;
