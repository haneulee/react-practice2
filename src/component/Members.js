import React, { Component } from "react";

const STATUS = {
  HOVERED: "hovered",
  NORMAL: "normal"
};

class Members extends Component {
  constructor(props) {
    super(props);
    this._onMouseEnter = this._onMouseEnter.bind(this);
    this._onMouseLeave = this._onMouseLeave.bind(this);
    this.state = {
      class: STATUS.NORMAL
    };
  }
  _onMouseEnter() {
    this.setState({ class: STATUS.HOVERED });
  }
  _onMouseLeave() {
    this.setState({ class: STATUS.NORMAL });
  }
  render() {
    return (
      <div>
        {/* <h2>Members</h2> */}
        <a
          className={this.state.class}
          href={this.props.page || "#"}
          onMouseEnter={this._onMouseEnter}
          onMouseLeave={this._onMouseLeave}
        >
          {this.props.children}
        </a>
      </div>
    );
  }
}

// export let sum = (v1, v2) => {
//   return v1 + v2;
// };

export default Members;
