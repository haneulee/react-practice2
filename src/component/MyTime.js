import React, { Component } from "react";
import PropTypes from "prop-types";

class MyTime extends Component {
  render() {
    return (
      <div className="row">
        <div className="col--xs-6">
          <button
            className="btn btnPrimary"
            onClick={() => this.props.changeTime()}
          >
            check current time!
          </button>
        </div>
        <div className="col-xs-6">
          <h4>
            <span className="label label-default">
              {this.props.curTime.toLocaleString()}
            </span>
          </h4>
        </div>
      </div>
    );
  }
}

MyTime.propTypes = {
  curTime: PropTypes.object.isRequired,
  changeTime: PropTypes.func.isRequired
};

export default MyTime;
