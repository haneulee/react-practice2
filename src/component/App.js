import React, { Component } from "react";
import CountryList from "./CountryList";
import Todo from "./Todo";
import Footer from "./Footer";
import styles from "./styles";
import update from "immutability-helper";
import MyButton from "./MyButton";
// import Portal from "./Portal";
import { Portal } from "react-portal";
import Modal from "./Modal";
import axios from "axios";
import { Container } from "flux/utils";
import ItemStore from "../flux/ItemStore";
import ItemAction from "../flux/ItemAction";
import MyTime from "./MyTime";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      msg: "todo list",
      showModal: false
    };
  }
  componentDidMount() {
    axios
      .post("http://sample.bmaster.kro.kr/contacts", {
        name: "sky",
        tel: "010-0000-0000",
        address: "seoul"
      })
      .then(Response => {
        if (Response.data.status !== "success") {
          throw new Error("data add error");
        }
        console.log(Response);
        return axios.get(
          "http://sample.bmaster.kro.kr/contacts/" + Response.data.no
        );
      })
      .then(Response => {
        console.log(Response.data);
      })
      .catch(error => {
        console.log(error);
      });
  }
  toggleModalBox = () => {
    this.setState({ showModal: !this.state.showModal });
  };
  add = todo => {
    // let newList = update(this.state.todolist, {
    //   $push: [{ no: new Date().getTime(), todo: todo, done: false }]
    // });
    // this.setState({ todolist: newList });

    this.props.addTodo(todo);
  };
  delete = no => {
    // let index = this.state.todolist.findIndex(todo => todo.no === no);
    // let newList = update(this.state.todolist, {
    //   $splice: [[index, 1]]
    // });
    // this.setState({ todolist: newList });
    this.props.deleteTodo(no);
  };
  toggle = no => {
    // let index = this.state.todolist.findIndex(todo => todo.no === no);
    // let toggle = !this.state.todolist[index].done;
    // let newList = update(this.state.todolist, {
    //   [index]: {
    //     done: { $set: toggle }
    //   }
    // });
    // this.setState({ todolist: newList });
    this.props.toggleTodo(no);
  };
  render() {
    const { showModal } = this.state;
    return (
      <div className="container">
        <h1 style={styles.title}>{this.state.msg}</h1>
        <hr style={styles} />
        {/* <MyButton addItem={this.addItem.bind(this)}>add</MyButton> */}
        {/* <MyButton subtract={this.subtract}>subtract</MyButton> */}
        {/* <CountryList itemlist={this.state.itemlist} /> */}
        <Todo
          todolist={this.props.todolist}
          add={this.add}
          delete={this.delete}
          toggle={this.toggle}
          isLog="true"
        />
        <button onClick={this.toggleModalBox}>modal</button>
        <MyTime
          changeTime={this.props.changeTime}
          curTime={this.props.curTime}
        />
        <Portal node={document && document.getElementById("modal-area")}>
          <Modal
            showModal={showModal}
            header="modal title"
            toggleModalBox={this.toggleModalBox}
          >
            <p>모달 컨텐츠~~</p>
          </Modal>
        </Portal>
        <Footer />
      </div>
    );
  }
}

// container 고차 함수를 이용해 itemstore객체를 store로 등록하고
// itemstore의 상태 데이터를 app 컴포넌트의 state로 설정함
// App.getStores = () => [ItemStore];
// App.calculateState = prevState => {
//   return ItemStore.getState();
// };
// const AppContainer = Container.create(App);

export default App;
