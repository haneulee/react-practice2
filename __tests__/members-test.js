"use strict";
import React from "react";
import Members from "../src/component/Members";
import renderer from "react-test-renderer";
import { sum } from "../src/component/Members";
jest.autoMockOff();

// describe("sum", function() {
//   it("adds 1+2 equal 3", function() {
//     expect(sum(1, 2)).toBe(3);
//   });
// });

it("renders correctly", () => {
  const tree = renderer
    .create(<Members page="http://www.facebook.com">facebook</Members>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it("renders as an anchor when no page is set", () => {
  const tree = renderer.create(<Members>facebook</Members>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("properly escapes quotes", () => {
  const tree = renderer
    .create(<Members>{"\"Facebook\" \\' is \\ 'awesome'"}</Members>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it("changes the class when hovered", () => {
  const component = renderer.create(
    <Members page="http://www.facebook.com">facebook</Members>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props._onMouseEnter();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props._onMouseLeave();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
