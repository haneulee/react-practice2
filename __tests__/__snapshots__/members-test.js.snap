// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<div>
  <a
    _onMouseEnter={[Function]}
    _onMouseLeave={[Function]}
    className="normal"
    href="http://www.facebook.com"
  >
    facebook
  </a>
</div>
`;

exports[`properly escapes quotes 1`] = `
<div>
  <a
    _onMouseEnter={[Function]}
    _onMouseLeave={[Function]}
    className="normal"
    href="#"
  >
    "Facebook" \\' is \\ 'awesome'
  </a>
</div>
`;

exports[`renders as an anchor when no page is set 1`] = `
<div>
  <a
    _onMouseEnter={[Function]}
    _onMouseLeave={[Function]}
    className="normal"
    href="#"
  >
    facebook
  </a>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <a
    _onMouseEnter={[Function]}
    _onMouseLeave={[Function]}
    className="normal"
    href="http://www.facebook.com"
  >
    facebook
  </a>
</div>
`;
